File: CSES/Sorting and Searching/1631 - Reading Books/readingBooks.py
n = int(input())
b = [int(i) for i in input().split()]
b.sort(reverse=True)

sumb = sum(b)
dif = sumb - b[0]
 
if b[0] > dif:
    print(b[0] * 2)
else:
    print(sumb)

File: CSES/Sorting and Searching/1631 - Reading Books/readingBooks.cpp
#include <bits/stdc++.h>
using namespace std;

#define FOR(i,a,b) for (int i = a; i < b; i++)
#define ll long long
typedef vector<int> vi;
#define sortv(v) sort((v).begin(), (v).end())

// Tried to make it as readable as possible, which result in longer code.

int main() {
  int n; cin >> n;
  ll sum = 0;
  vi books(n);

  FOR(i,0,n){
    cin >> books[i];
    sum += books[i];
  }

  sortv(books);

  int bigger = books[n-1];
  ll dif = sum - bigger;
  
  if (dif < bigger){
    cout << 2 * bigger << endl;
  } else {
    cout << sum << endl;
  }
}
